### PR TITLE
Update DuckDB to latest

### PR DESCRIPTION
**Pull request by majetideepak**

### CODE REVIEW

**Comment by majetideepak** [comment]
@Mytherin Looks like there are a couple of issues with the amalgamation with the latest DuckDB branch. See CI run below. Do you want me to file Issues for them? Thanks.
https://app.circleci.com/pipelines/github/facebookincubator/velox/3218/workflows/edfa3fa2-b809-4510-8ef6-df4ad243299b/jobs/14887
CC: @mbasmanova

**Comment by Mytherin** [comment]
I have pushed a fix for the compilation failures relating to the C++17 warnings in duckdb/duckdb#2744.

The build failures in `DuckParser.cpp` require patching on your side, however. These are caused by a change to the `ColumnRefExpression` in the context of duckdb/duckdb#2578.

ColumnRefExpressions used to only hold a `table_name` and a `column_name`, but now hold a `vector<string> column_names` to account for long chains of dots (e.g. `col.field1.field2.field3...`).

You can use the `GetColumnName()` and `GetTableName()` helper functions to retrieve the table/column names as before, but note that `GetTableName()` requires `column_names` to have at 2 members.

**Comment by majetideepak** [comment]
Release mode build is throwing. I will take a look.
```
[----------] 16 tests from HashJoinTest
[ RUN      ] HashJoinTest.bigintArray
../../velox/exec/tests/utils/QueryAssertions.cpp:413: Failure
Value of: duckDbResult->success
  Actual: false
Expected: true
DuckDB query failed: Invalid Input Error: Type INT64 with value -1959926401839265 can't be cast because the value is out of range for the destination type UINT64
SELECT t_k0, t_data, u_k0, u_data FROM   t, u   WHERE t_k0 = u_k0
../../velox/exec/tests/utils/QueryAssertions.cpp:470: Failure
Value of: false
  Actual: false
Expected: true
```

**Comment by mbasmanova** [comment]
> This PR also adds a gitattributes file for the vendored DuckDB code.

@majetideepak What does this do?

**Comment by mbasmanova** [comment]
@majetideepak It would be nice to following https://cbea.ms/git-commit/ guide when writing PR titles and description. The PR title here is great, but the description could use some editing.

```
- Update DuckDB to the latest version.
- Also add a gitattributes file for the vendored DuckDB code.
```

**Comment by majetideepak** [comment]
> > This PR also adds a gitattributes file for the vendored DuckDB code.
> 
> @majetideepak What does this do?

We should ideally not count generated, vendored code towards the project language statistics. gitattributes can be used to indicate such generated and vendored code. see  https://github.com/github/linguist/blob/master/docs/overrides.md
Also, see https://github.com/facebookexternal/presto_cpp/blob/main/presto_cpp/presto_protocol/.gitattributes

**Comment by majetideepak** [comment]
> @majetideepak It would be nice to following https://cbea.ms/git-commit/ guide when writing PR titles and description. The PR title here is great, but the description could use some editing.

I get confused sometimes with this. The guide says to start the description with a subject "Summarize changes in around 50 characters or less". Then in another paragraph, we can expand or add other changes. So I put the main change as the subject and put the gitattributes change as a bullet.

**Comment by mbasmanova** [comment]
> > @majetideepak It would be nice to following https://cbea.ms/git-commit/ guide when writing PR titles and description. The PR title here is great, but the description could use some editing.
> 
> I get confused sometimes with this. The guide says to start the description with a subject "Summarize changes in around 50 characters or less". Then in another paragraph, we can expand or add other changes. So I put the main change as the subject and modified the gitattributes change as a bullet.

Ideally, each PR would include only one logical change. In this case, I'd move gitattributes to a separate PR.

**Comment by mbasmanova** [comment]
> > > This PR also adds a gitattributes file for the vendored DuckDB code.
> > 
> > 
> > @majetideepak What does this do?
> 
> We should ideally not count generated, vendored code towards the project language statistics. gitattributes can be used to indicate such generated and vendored code. see https://github.com/github/linguist/blob/master/docs/overrides.md Also, see https://github.com/facebookexternal/presto_cpp/blob/main/presto_cpp/presto_protocol/.gitattributes

Thank you for the explanation. Sounds like a useful change. It would be nice to put it in a separate PR and explain its effect in that PR's description.

**Comment by majetideepak** [comment]
> Thank you for the explanation. Sounds like a useful change. It would be nice to put it in a separate PR and explain its effect in that PR's description.

sounds good. I will move this change.

**Comment by majetideepak** [comment]
I also realized in the Github UI, the title is the subject.

**Comment by majetideepak** [comment]
@Mytherin I am hitting another error after updating to a recent DuckDB version.
CI Build: https://app.circleci.com/pipelines/github/facebookincubator/velox/3444/workflows/0798db86-66e9-498b-8734-3fcd772ccea0/jobs/16193
Some details below:
```
112: Value of: duckDbResult->success
112:   Actual: false
112: Expected: true
112: DuckDB query failed: Invalid Input Error: Type INT64 with value -1959926401839265 can't be cast because the value is out of range for the destination type UINT64
112: SELECT t_k0, t_data, u_k0, u_data FROM   t, u   WHERE t_k0 = u_k0
112: ../../velox/exec/tests/utils/QueryAssertions.cpp:470: Failure
```
I did some investigation and looks like a cast from `int64_t` to `uint64_t` is failing. Below is the bt. Can you help with this? Thanks.
```
 frame #0: 0x00007ff801444ca8 libc++abi.dylib`__cxa_throw
    frame #1: 0x00000001093d98a6 velox_exec_test`unsigned long long duckdb::Value::GetValueInternal<unsigned long long>() const [inlined] unsigned long long duckdb::Cast::Operation<long long, unsigned long long>(input=-5964650651471824) at duckdb.cpp:16685:4 [opt]
    frame #2: 0x00000001093d9845 velox_exec_test`unsigned long long duckdb::Value::GetValueInternal<unsigned long long>(this=0x00007ff7b7c07e88) const at duckdb.cpp:34828 [opt]
    frame #3: 0x00000001095be857 velox_exec_test`duckdb::CheckForPerfectJoinOpt(duckdb::LogicalComparisonJoin&, duckdb::PerfectHashJoinStats&) [inlined] unsigned long long duckdb::Value::GetValue<unsigned long long>(this=<unavailable>) const at duckdb.cpp:34919:9 [opt]
    frame #4: 0x00000001095be852 velox_exec_test`duckdb::CheckForPerfectJoinOpt(op=0x0000600001041950, join_state=0x00007ff7b7c07fb0) at duckdb.cpp:62480 [opt]
    frame #5: 0x00000001095bf461 velox_exec_test`duckdb::PhysicalPlanGenerator::CreatePlan(this=0x00007ff7b7c08680, op=0x0000600001041950) at duckdb.cpp:62584:3 [opt]
    frame #6: 0x00000001095bc942 velox_exec_test`duckdb::PhysicalPlanGenerator::CreatePlan(this=0x00007ff7b7c08680, op=0x0000600001041950) at duckdb.cpp:65445:10 [opt]
    frame #7: 0x00000001095c70b3 velox_exec_test`duckdb::PhysicalPlanGenerator::CreatePlan(this=0x00007ff7b7c08680, op=0x0000600000455260) at duckdb.cpp:64571:14 [opt]
    frame #8: 0x00000001095bcc14 velox_exec_test`duckdb::PhysicalPlanGenerator::CreatePlan(this=0x00007ff7b7c08680, op=0x0000600000455260) at duckdb.cpp:65415:10 [opt]
    frame #9: 0x00000001095c5c1b velox_exec_test`duckdb::PhysicalPlanGenerator::CreatePlan(this=0x00007ff7b7c08680, op=<unavailable>) at duckdb.cpp:65404:14 [opt]
    frame #10: 0x0000000109717269 velox_exec_test`duckdb::ClientContext::CreatePreparedStatement(this=0x00007f8148826218, lock=<unavailable>, query=<unavailable>, statement=<unavailable>) at duckdb.cpp:97670:40 [opt]
    frame #11: 0x000000010971beb1 velox_exec_test`duckdb::ClientContext::RunStatementInternal(this=0x00007f8148826218, lock=<unavailable>, query=<unavailable>, statement=nullptr, allow_stream_result=false) at duckdb.cpp:97860:18 [opt]
    frame #12: 0x000000010971b63a velox_exec_test`duckdb::ClientContext::RunStatementOrPreparedStatement(this=<unavailable>, lock=0x0000600002de4260, query=<unavailable>, statement=nullptr, prepared=nullptr, values=0x0000000000000000, allow_stream_result=false) at duckdb.cpp:97912:13 [opt]
    frame #13: 0x000000010971d8ff velox_exec_test`duckdb::ClientContext::RunStatement(this=0x00007f8148826218, lock=0x0000600002de4260, query="SELECT t_k0, t_data, u_k0, u_data FROM   t, u   WHERE t_k0 = u_k0", statement=nullptr, allow_stream_result=<unavailable>) at duckdb.cpp:97962:9 [opt]
```

**Comment by Mytherin** [comment]
Thanks for the report, this was indeed a regression caused by an optimizer failing when `max - min` did not fit into an INT64. I have pushed a fix in duckdb/duckdb#2798. Let me know if that fixes the problem for you.

**Comment by majetideepak** [comment]
@Mytherin Thanks for all the help with this PR. Everything now looks good.
@mbasmanova Can you import this and let me know if there are other failures? Thanks.

**Review by mbasmanova — APPROVED** [review]

**Comment by mbasmanova** [comment]
@majetideepak I'm getting build errors:

```
stderr: In file included from velox/dwio/parquet/reader/duckdb/InputStreamFileHandle.cpp:18:
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:39:58: error: non-virtual member function marked 'override' hides virtual member function
          ::duckdb::FileCompressionType::UNCOMPRESSED*/) override {
                                                         ^
velox/external/duckdb/duckdb.hpp:1853:33: note: hidden overloaded virtual function 'duckdb::FileSystem::OpenFile' declared here: different number of parameters (5 vs 4)
        virtual unique_ptr<FileHandle> OpenFile(const string &path, uint8_t flags, FileLockType lock = DEFAULT_LOCK,
                                       ^
In file included from velox/dwio/parquet/reader/duckdb/InputStreamFileHandle.cpp:18:
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:167:31: error: only virtual member functions can be marked 'override'
  std::string PathSeparator() override {
                              ^~~~~~~~~
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:172:7: error: only virtual member functions can be marked 'override'
      override {
      ^~~~~~~~~
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:176:62: error: only virtual member functions can be marked 'override'
  std::string ConvertSeparators(const std::string& /*path*/) override {
                                                             ^~~~~~~~~
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:180:60: error: only virtual member functions can be marked 'override'
  std::string ExtractBaseName(const std::string& /*path*/) override {
                                                           ^~~~~~~~~
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:188:57: error: only virtual member functions can be marked 'override'
  void SetWorkingDirectory(const std::string& /*path*/) override {
                                                        ^~~~~~~~~
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:192:37: error: only virtual member functions can be marked 'override'
  std::string GetWorkingDirectory() override {
                                    ^~~~~~~~~
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:196:34: error: only virtual member functions can be marked 'override'
  std::string GetHomeDirectory() override {
                                 ^~~~~~~~~
velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:204:33: error: only virtual member functions can be marked 'override'
  uint64_t GetAvailableMemory() override {
                                ^~~~~~~~~
9 errors generated.
    When running <c++ preprocess_and_compile>.
    When building rule //velox/dwio/parquet/reader/duckdb:velox_dwio_parquet_reader_duckdb#compile-pic-InputStreamFileHandle.cpp.oa239b50a,platform009-clang (ovr_config//platform/linux:x86_64-fbcode).
```

**Comment by Mytherin** [comment]
Most of those methods have been made static and the implementations there look like stubs, so they can probably just be removed. The error with `OpenFile` seems to be down to adding another parameter to this interface (`FileOpener *opener`). You should be able to just add and ignore this parameter.

**Comment by majetideepak** [comment]
@mbasmanova The errors should be fixed now. I will update the CircleCI adapters run to include Parquet in another PR.

**Comment by mbasmanova** [comment]
> @mbasmanova The errors should be fixed now. I will update the CircleCI adapters run to include Parquet in another PR.

@majetideepak Thank you, Deepak. Let me try again.

**Comment by mbasmanova** [comment]
@majetideepak I'm seeing more failures. Here is an example:

```
Failed to parse output from the test suite: IO error: No result xml files found.
Execution directory: /tmp/tpx-20211216-052954.783100/4cb3b06a-08d4-4c4d-b8ce-8021d1afa689
stdout:
Note: Google Test filter = MapTest.noNulls
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from MapTest
[ RUN      ] MapTest.noNulls

stderr:
terminate called after throwing an instance of 'duckdb::InternalException'
  what():  INTERNAL Error: Transformer::StackCheck variables are incorrectly set up
*** Aborted at 1639661644 (Unix time, try 'date -d @1639661644') ***
*** Signal 6 (SIGABRT) (0x755900002ff8) received by PID 12280 (pthread TID 0x7f0b37b68f40) (linux TID 12280) (maybe from PID 12280, UID 30041) (code: -6), stack trace: ***
    @ 00000000000192fd folly::symbolizer::(anonymous namespace)::innerSignalHandler(int, siginfo_t*, void*)
                       ./folly/experimental/symbolizer/SignalHandler.cpp:449
    @ 00000000000102ab folly::symbolizer::(anonymous namespace)::signalHandler(int, siginfo_t*, void*)
                       ./folly/experimental/symbolizer/SignalHandler.cpp:470
    @ 0000000000000000 (unknown)
    @ 000000000003e530 __GI_raise
    @ 000000000002551c __GI_abort
    @ 00000000000a07d4 __gnu_cxx::__verbose_terminate_handler()
    @ 000000000009e4a5 __cxxabiv1::__terminate(void (*)())
    @ 000000000009e510 std::terminate()
    @ 000000000009e7db __cxa_throw
    @ 0000000001b02472 duckdb::Transformer::StackCheck(unsigned long)
                       ./velox/external/duckdb/duckdb.cpp:139795
    @ 0000000001b3559b duckdb::Transformer::TransformSelectNode(duckdb_libpgquery::PGSelectStmt*)
                       ./velox/external/duckdb/duckdb.cpp:139190
    @ 0000000001b22fc5 duckdb::Transformer::TransformSelect(duckdb_libpgquery::PGNode*, bool)
                       ./velox/external/duckdb/duckdb.cpp:139173
    @ 0000000001b567fd duckdb::Transformer::TransformStatement(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:139817
    @ 0000000001b56663 duckdb::Transformer::TransformStatement(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:139809
    @ 0000000001adba4d duckdb::Transformer::TransformParseTree(duckdb_libpgquery::PGList*, std::vector<std::unique_ptr<duckdb::SQLStatement, std::default_delete<duckdb::SQLStatement> >, std::allocator<std::unique_ptr<duckdb::SQLStatement, std::default_delete<duckdb::SQLStatement> > > >&)
                       ./velox/external/duckdb/duckdb.cpp:139777
    @ 0000000001956777 duckdb::Parser::ParseQuery(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/external/duckdb/duckdb.cpp:133481
    @ 00000000019bc5cf duckdb::Parser::ParseExpressionList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/external/duckdb/duckdb.cpp:133538
    @ 000000000007227b facebook::velox::duckdb::parseExpr(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/duckdb/conversion/DuckParser.cpp:370
    @ 000000000000b122 facebook::velox::parse::parseExpr(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/parse/ExpressionsParser.cpp:24
    @ 00000000006f6669 facebook::velox::functions::test::FunctionBaseTest::makeTypedExpr(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<facebook::velox::RowType const> const&)
                       ./velox/functions/prestosql/tests/FunctionBaseTest.h:181
                       -> ./velox/functions/prestosql/tests/ArithmeticTest.cpp
    @ 00000000006f5216 facebook::velox::functions::test::FunctionBaseTest::evaluate(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<facebook::velox::RowVector> const&)
                       ./velox/functions/prestosql/tests/FunctionBaseTest.h:486
                       -> ./velox/functions/prestosql/tests/ArithmeticTest.cpp
    @ 0000000000e9a76c std::shared_ptr<facebook::velox::MapVector> facebook::velox::functions::test::FunctionBaseTest::evaluate<facebook::velox::MapVector>(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::shared_ptr<facebook::velox::RowVector> const&)
                       ./velox/functions/prestosql/tests/FunctionBaseTest.h:499
                       -> ./velox/functions/prestosql/tests/MapConcatTest.cpp
    @ 0000000000ef31e7 MapTest_noNulls_Test::TestBody()
                       ./velox/functions/prestosql/tests/MapTest.cpp:38
    @ 000000000004d6a5 testing::Test::Run()
                       /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest.cc:2680
                       -> /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest-all.cc
    @ 000000000004d814 testing::TestInfo::Run()
                       /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest.cc:2857
                       -> /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest-all.cc
    @ 000000000004de71 testing::TestSuite::Run()
                       /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest.cc:3011
                       -> /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest-all.cc
    @ 000000000004e63e testing::internal::UnitTestImpl::RunAllTests()
                       /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest.cc:5728
                       -> /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest-all.cc
    @ 000000000004d937 testing::UnitTest::Run()
                       /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest.cc:2663
                       -> /home/engshare/third-party2/googletest/20201023/src/googletest/googletest/src/gtest-all.cc
    @ 0000000000002680 RUN_ALL_TESTS()
                       ./third-party-buck/platform009/build/googletest/include/gtest/gtest.h:2486
                       -> ./common/gtest/LightMain.cpp
    @ 00000000000022a4 main
                       ./common/gtest/LightMain.cpp:20
    @ 0000000000025dc4 __libc_start_main
    @ 000000000065b029 _start
                       /home/engshare/third-party2/glibc/2.30/src/glibc-2.30/csu/../sysdeps/x86_64/start.S:120

exit_code:"Exit code from the process: 134"
```

**Comment by mbasmanova** [comment]
Screen Shot 2021-12-16 at 11 58 11 AM

**Comment by mbasmanova** [comment]
Also, seeing linter failures:

Screen Shot 2021-12-16 at 11 58 39 AM
Screen Shot 2021-12-16 at 11 58 53 AM

**Comment by mbasmanova** [comment]
@Mytherin Mark, we are getting some linter errors and test failures with the latest duckdb. Any chance you could take a look?

**Comment by majetideepak** [comment]
@mbasmanova I updated the patch. Can you try now?

**Comment by mbasmanova** [comment]
@Mytherin @majetideepak Mark, Deepak, I still see failures:

```
  what():  Parser Error: Stack usage in parsing is too high: the query tree is too deep (stack usage 1012096, max stack usage 102400)
*** Aborted at 1639757654 (Unix time, try 'date -d @1639757654') ***
*** Signal 6 (SIGABRT) (0x7559000025c0) received by PID 9664 (pthread TID 0x7effac54b000) (linux TID 9664) (maybe from PID 9664, UID 30041) (code: -6), stack trace: ***
    @ 00000000000192fd folly::symbolizer::(anonymous namespace)::innerSignalHandler(int, siginfo_t*, void*)
                       ./folly/experimental/symbolizer/SignalHandler.cpp:449
    @ 00000000000102ab folly::symbolizer::(anonymous namespace)::signalHandler(int, siginfo_t*, void*)
                       ./folly/experimental/symbolizer/SignalHandler.cpp:470
    @ 0000000000000000 (unknown)
    @ 000000000003e530 __GI_raise
    @ 000000000002551c __GI_abort
    @ 00000000000a07d4 __gnu_cxx::__verbose_terminate_handler()
    @ 000000000009e4a5 __cxxabiv1::__terminate(void (*)())
    @ 000000000009e510 std::terminate()
    @ 000000000009e7db __cxa_throw
    @ 0000000001b1596c duckdb::Transformer::StackCheck(unsigned long)
                       ./velox/external/duckdb/duckdb.cpp:140236
    @ 0000000001b108ca duckdb::Transformer::TransformExpression(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:136138
    @ 0000000001b25f52 duckdb::Transformer::TransformFuncCall(duckdb_libpgquery::PGFuncCall*)
                       ./velox/external/duckdb/duckdb.cpp:136403
    @ 0000000001b10b41 duckdb::Transformer::TransformExpression(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:136148
    @ 0000000001b22d6f duckdb::Transformer::TransformAExpr(duckdb_libpgquery::PGAExpr*)
                       ./velox/external/duckdb/duckdb.cpp:136926
    @ 0000000001b10b22 duckdb::Transformer::TransformExpression(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:136146
    @ 0000000001b1f96f duckdb::Transformer::TransformResTarget(duckdb_libpgquery::PGResTarget*)
                       ./velox/external/duckdb/duckdb.cpp:136113
    @ 0000000001b10c1a duckdb::Transformer::TransformExpression(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:136162
    @ 0000000001b2ce5f duckdb::Transformer::TransformExpressionList(duckdb_libpgquery::PGList&, std::vector<std::unique_ptr<duckdb::ParsedExpression, std::default_delete<duckdb::ParsedExpression> >, std::allocator<std::unique_ptr<duckdb::ParsedExpression, std::default_delete<duckdb::ParsedExpression> > > >&)
                       ./velox/external/duckdb/duckdb.cpp:136194
    @ 0000000001b49692 duckdb::Transformer::TransformSelectNode(duckdb_libpgquery::PGSelectStmt*)
                       ./velox/external/duckdb/duckdb.cpp:139678
    @ 0000000001b36325 duckdb::Transformer::TransformSelect(duckdb_libpgquery::PGNode*, bool)
                       ./velox/external/duckdb/duckdb.cpp:139612
    @ 0000000001b69b5d duckdb::Transformer::TransformStatement(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:140254
    @ 0000000001b699c3 duckdb::Transformer::TransformStatement(duckdb_libpgquery::PGNode*)
                       ./velox/external/duckdb/duckdb.cpp:140246
    @ 0000000001aeef2d duckdb::Transformer::TransformParseTree(duckdb_libpgquery::PGList*, std::vector<std::unique_ptr<duckdb::SQLStatement, std::default_delete<duckdb::SQLStatement> >, std::allocator<std::unique_ptr<duckdb::SQLStatement, std::default_delete<duckdb::SQLStatement> > > >&)
                       ./velox/external/duckdb/duckdb.cpp:140216
    @ 0000000001969c47 duckdb::Parser::ParseQuery(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/external/duckdb/duckdb.cpp:133920
    @ 00000000019cfaaf duckdb::Parser::ParseExpressionList(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/external/duckdb/duckdb.cpp:133977
    @ 000000000007227b facebook::velox::duckdb::parseExpr(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/duckdb/conversion/DuckParser.cpp:370
    @ 000000000000b122 facebook::velox::parse::parseExpr(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)
                       ./velox/parse/ExpressionsParser.cpp:24
```

**Comment by majetideepak** [comment]
I confirmed that the patch https://github.com/duckdb/duckdb/pull/2808 is included. See raw file below
https://raw.githubusercontent.com/facebookincubator/velox/1d6e1b5526da1273a3305f6012de4d5263011b51/velox/external/duckdb/duckdb.cpp

**Comment by majetideepak** [comment]
The error here is different. `Stack usage in parsing is too high: the query tree is too deep (stack usage 1012096, max stack usage 102400)`

**Comment by Mytherin** [comment]
I have updated duckdb/duckdb#2808 to revert to our old way of stack overflow prevention (expression tree depth limit), which should fix the stack issue. The linter failures are outdated now, however, so the line numbers no longer match up which makes them hard to fix. Could you re-run and post the updated line numbers? Then I can have a go at fixing them.

Also is there any way of running these linters ourselves in our CI, or are these linters not open sourced?

**Comment by mbasmanova** [comment]
@Mytherin Thank you, Mark.

> Also is there any way of running these linters ourselves in our CI, or are these linters not open sourced?

These linters are not open sourced. I'll make a note to look into open sourcing these. CC: @kgpai @pedroerp @spershin @amitkdutta

**Comment by mbasmanova** [comment]
@Mytherin Mark, here are the MoveEvaluationOrder failures on the latest version:

Screen Shot 2021-12-20 at 3 56 10 PM

facebook-hte-StdRegexIsAwful are still there as well.

**Comment by Mytherin** [comment]
Should all be fixed in duckdb/duckdb#2837. The define `DISABLE_DUCKDB_REMOTE_INSTALL` has to be added to disable the use of `std::regex`.

**Comment by mbasmanova** [comment]
> Should all be fixed in [duckdb/duckdb#2837](https://github.com/duckdb/duckdb/pull/2837). The define `DISABLE_DUCKDB_REMOTE_INSTALL` has to be added to disable the use of `std::regex`.

Thank you, Mark.

**Comment by mbasmanova** [comment]
@majetideepak @Mytherin I still get StdRegexIsAwful lint error

Screen Shot 2021-12-23 at 1 46 25 PM
.

**Comment by mbasmanova** [comment]
@majetideepak Deepak, I see that you added

```
add_compile_definitions(DISABLE_DUCKDB_REMOTE_INSTALL)
```

I suppose I need to add something similar to buck TARGETS files.

File: velox/external/duckdb/CMakeLists.txt [hunk 1]
```diff
@@ -13,4 +13,6 @@ add_subdirectory(tpch)
 
 add_library(duckdb duckdb.cpp parquet-amalgamation.cpp)
 
+add_compile_definitions(DISABLE_DUCKDB_REMOTE_INSTALL)
```
Comment (Mytherin) [inline velox/external/duckdb/CMakeLists.txt:16]: I suspect this might need to be moved up, above the `add_library` that defines the source files.

Comment (majetideepak) [reply velox/external/duckdb/CMakeLists.txt:16]: I don't think the order matters. I see that the generated make file has the definition locally.
```
/Users/deepak/workspace/ahana/velox/cmake-build-debugwiths3installed/velox/external/duckdb && /Library/Developer/CommandLineTools/usr/bin/c++ -DCODEGEN_ENABLED=1 -DDISABLE_DUCKDB_REMOTE_INSTALL -DVELOX_ENABLE_S3 -I/usr/local/opt/icu4c/include -I/Users/deepak/workspace/ahana/velox/. -isystem /usr/local/include -isystem /Users/deepak/workspace/ahana/velox/velox -isystem /Users/deepak/workspace/ahana/velox/velox/external -isystem /Users/deepak/workspace/ahana/velox/velox/external/duckdb -mavx2 -mfma -mavx -mf16c -mbmi2 -march=native -D USE_VELOX_COMMON_BASE -D HAS_UNCAUGHT_EXCEPTIONS -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -std=gnu++17 -o CMakeFiles/duckdb.dir/duckdb.cpp.o -c /Users/deepak/workspace/ahana/velox/velox/external/duckdb/duckdb.cpp
```
@mbasmanova can you try updating the buck scripts?

**Comment by mbasmanova** [comment]
I update the TARGETS file and the error went away. Thank you @majetideepak and @Mytherin for getting this though.
